### DISCUSSION TITLE
Sort table on model attribute not in a DB column.

### DISCUSSION THREAD

**Discussion by lucacastelnuovo** [filamentphp/filament#6974] [Help]
I've created a custom attribute on a model.

```php
// ...
     public function isMain(): Attribute
    {
        return Attribute::make(
            get: fn (): bool => $this->relation->value,
        );
    }
```

Can I sort a table to make sure the main record shows up on top?

**Comment by danharrin** [comment]
No, you should try and move it into your DB layer. Sorting in PHP across many records would not be performant.

**Reply by lucacastelnuovo** [reply]
You are right, do not know how to do it yet. But thank you for pointing me in the right direction.
